test/operators-spec.js

```diff
@@ -52,5 +52,82 @@ describe('operators', () => {
       expect(lulz).to.deep.equal([{ type: 'LULZ', i: 0 }, { type: 'LARF', i: 1 }]);
       expect(haha).to.deep.equal([{ type: 'HAHA', i: 0 }]);
     });
+
+    it('should handle actionCreators which define a toString method', () => {
+      // This helps when using the popular `redux-actions` npm package.
+      const actions = new Subject();
+      const cache1 = [];
+      const cache2 = [];
+      const LULZ_TYPE = 'LULZ';
+      const HAHA_TYPE = 'HAHA';
+      const LARF_TYPE = Symbol();
+
+      const createActionCreator = type => {
+        const actionCreator = payload => ({ type, payload });
+        actionCreator.toString = () => type;
+        return actionCreator;
+      };
+
+      const lulz = createActionCreator(LULZ_TYPE);
+      const haha = createActionCreator(HAHA_TYPE);
+      const larf = createActionCreator(LARF_TYPE);
+
+      // Sanity check.
+      expect(String(lulz)).to.deep.equal(LULZ_TYPE);
+      expect(lulz(0)).to.deep.equal({ type: LULZ_TYPE, payload: 0 });
+
+      actions.pipe(ofType(lulz, larf)).subscribe(x => cache1.push(x));
+      actions.pipe(ofType(haha)).subscribe(x => cache2.push(x));
+
+      actions.next(lulz(0));
+      expect(cache1).to.deep.equal([lulz(0)]);
+      expect(cache2).to.deep.equal([]);
+
+      actions.next(larf(1));
+      expect(cache1).to.deep.equal([lulz(0), larf(1)]);
+      expect(cache2).to.deep.equal([]);
+
+      actions.next(haha(0));
+      expect(cache1).to.deep.equal([lulz(0), larf(1)]);
+      expect(cache2).to.deep.equal([haha(0)]);
+    });
+
+    it('should not fail when types are symbols', () => {
+      const actions = new Subject();
+      const cache1 = [];
+      const cache2 = [];
+      const LULZ_TYPE = Symbol();
+      const HAHA_TYPE = Symbol();
+      const LARF_TYPE = Symbol();
+
+      // Sanity check.
+      expect(HAHA_TYPE).not.to.equal(LULZ_TYPE);
+      expect(LULZ_TYPE).not.to.equal(LARF_TYPE);
+      expect(LARF_TYPE).not.to.equal(HAHA_TYPE);
+      expect(HAHA_TYPE.toString()).to.equal(LULZ_TYPE.toString());
+      expect(LULZ_TYPE.toString()).to.equal(LARF_TYPE.toString());
+      expect(LARF_TYPE.toString()).to.equal(HAHA_TYPE.toString());
+      expect(String(HAHA_TYPE)).to.equal(String(LULZ_TYPE));
+      expect(String(LULZ_TYPE)).to.equal(String(LARF_TYPE));
+      expect(String(LARF_TYPE)).to.equal(String(HAHA_TYPE));
```

Comment:
I was surprised to see that `String(Symbol())` doesn't throw an error. But...

```
const foo = () => {};
foo.toString = () => Symbol();
String(foo);
```

DOES throw an error. Is this something that I'm not understanding that I need to take into account?

Reply:
`toString` should always return a string. If someone overrides it and does not return a string, that's their fault lol 😄